scripts/lib/CIME/case_st_archive.py

```diff
@@ -20,45 +20,143 @@ def _get_archive_file_fn(copy_only):
     """
     return shutil.copyfile if copy_only else shutil.move
 
-
 ###############################################################################
-def _get_datenames(case, last_date=None):
+def _get_datenames(rundir, casename):
 ###############################################################################
     """
-    Returns a list of datenames giving the dates of cpl restart files
-
-    If there are no cpl restart files, this will return []
+    Returns the datetime objects specifying the times of each file
+    Note we are assuming that the coupler restart files exist and are consistent with other component datenames
```

Comment:
@mfdeakin-sandia made a lot of changes to this file.

Reply:
@mfdeakin-sandia please help

Reply:
I'll look into it - are there tests for the multi driver code, so I can make certain I put them back correctly?

Reply:
yes, MCC in cime_developer is one.